core/controllers/collection_editor.py

```diff
@@ -155,6 +140,25 @@ def _require_valid_version(self, version_from_payload, collection_version):
                 % (collection_version, version_from_payload))
 
     @require_editor
+    def get(self, collection_id):
+        """Populates the data on the individual collection page."""
+
+        try:
+            # Try to retrieve collection
+            collection_dict = (
+                summary_services.get_learner_collection_dict_by_id(
+                    collection_id, self.user_id,
+                    allow_invalid_explorations=True))
+        except Exception as e:
+            raise self.PageNotFoundException(e)
+
+        self.values.update({
+            'collection': collection_dict
```

Comment:
Discussed at TL meeting; you bring up great points. Let's keep it as-is because it is better to introduce a new pattern all-out rather than as part of a random PR. Thanks!